### PR TITLE
Added VoxPopuli Datasets

### PR DESCRIPTION
**Pull request by AdnanElAssadi56**
### Code Quality

- [ ] **Code Formatted**: Format the code using `make lint` to maintain consistent style.

### Documentation

- [ ] **Updated Documentation**: Add or update documentation to reflect the changes introduced in this PR.

### Testing

- [ ] **New Tests Added**: Write tests to cover new functionality. Validate with `make test-with-coverage`.
- [ ] **Tests Passed**: Run tests locally using `make test` or `make test-with-coverage` to ensure no existing functionality is broken.

### Adding datasets checklist

**Reason for dataset addition**: ... 

- [ ] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [ ] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [ ] `intfloat/multilingual-e5-small`
- [ ] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [ ] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [ ] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [ ] Run tests locally to make sure nothing is broken using `make test`.
- [ ] Run the formatter to format the code using `make lint`.

### Adding a model checklist

 - [ ] I have filled out the ModelMeta object to the extent possible
 - [ ] I have ensured that my model can be loaded using
   - [ ] `mteb.get_model(model_name, revision)` and
   - [ ] `mteb.get_model_meta(model_name, revision)`
 - [ ] I have tested the implementation works on a representative set of tasks.

### CODE REVIEW

**Comment by KennethEnevoldsen** [comment]
See comment on: #2641

File: mteb/tasks/Audio/Clustering/eng/VoxPopuliGenderClustering.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        description="Clustering speech samples by speaker gender (male/female) from European Parliament recordings.",
+        reference="https://huggingface.co/datasets/facebook/voxpopuli",
+        dataset={
+            "path": "facebook/voxpopuli",
+            "name": "en",  # This selects the english config
+            "revision": "719aaef8225945c0d80b277de6c79aa42ab053d5",
+        },
+        type="AudioClustering",
+        category="a2a",
+        eval_splits=["validation", "test"],
```
Comment (isaac-chung) [inline mteb/tasks/Audio/Clustering/eng/VoxPopuliGenderClustering.py]: Let's only do test.
```suggestion
        eval_splits=["test"],
```

File: mteb/tasks/Audio/Clustering/eng/VoxPopuliGenderClustering.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        eval_langs=["eng-Latn"],  # Focus on one language for clustering
```
Comment (isaac-chung) [inline mteb/tasks/Audio/Clustering/eng/VoxPopuliGenderClustering.py]: Why focus only on one language, if this dataset comes in multilingual form?

File: mteb/tasks/Audio/Clustering/eng/VoxPopuliAccentClustering.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+    audio_column_name: str = "audio"
+
+    def dataset_transform(self):
+        # Split test into train (80%) and new test (20%)
+        import random
+
+        import numpy as np
+
+        random.seed(42)
```
Comment (isaac-chung) [inline mteb/tasks/Audio/Clustering/eng/VoxPopuliAccentClustering.py]: AbsTask has a self.seed attribute.

File: mteb/tasks/Audio/AudioClassification/eng/VoxPopuliLanguageID.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        },
+    )
+
+    audio_column_name: str = "audio"
+    label_column_name: str = "language"
+    samples_per_label: int = 50  # For balanced training
+    is_cross_validation: bool = False
+
+    def dataset_transform(self):
+        """Create train and test splits from the original test split."""
```
Comment (isaac-chung) [inline mteb/tasks/Audio/AudioClassification/eng/VoxPopuliLanguageID.py]: Why not just take the original test split?

File: mteb/tasks/Audio/Clustering/eng/VoxPopuliAccentClustering.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        train_indices = indices[:split_point]
+        test_indices = indices[split_point:]
+
+        self.dataset = {
+            "train": test_data.select(train_indices),
+            "test": test_data.select(test_indices),
+        }
+        print(
+            f"Created train split with {len(train_indices)} samples and test split with {len(test_indices)} samples"
+        )
```
Comment (isaac-chung) [inline mteb/tasks/Audio/Clustering/eng/VoxPopuliAccentClustering.py]: I don't think train set is needed in clustering right? We've always just [fed in the test split only.](https://github.com/embeddings-benchmark/mteb/blob/maeb/mteb/tasks/Image/ImageClustering/eng/CIFAR.py)

File: mteb/tasks/Audio/AudioClassification/eng/VoxPopuliGenderID.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        description="Classification of speech samples by speaker gender (male/female) from European Parliament recordings.",
+        reference="https://huggingface.co/datasets/facebook/voxpopuli",
+        dataset={
+            "path": "facebook/voxpopuli",
+            "name": "en",  # This selects the english config/subset
+            "revision": "719aaef8225945c0d80b277de6c79aa42ab053d5",
+        },
+        type="AudioClassification",
+        category="a2t",
+        eval_splits=["validation", "test"],
```
Comment (isaac-chung) [inline mteb/tasks/Audio/AudioClassification/eng/VoxPopuliGenderID.py]: Let's only take the test split.
```suggestion
        eval_splits=["test"],
```

File: mteb/tasks/Audio/AudioClassification/eng/VoxPopuliAccentID.py [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+        description="Classification of English speech samples into one of 15 non-native accents from European Parliament recordings.",
+        reference="https://huggingface.co/datasets/facebook/voxpopuli",
+        dataset={
+            "path": "facebook/voxpopuli",
+            "name": "en_accented",  # This explicitly selects the accented English config
+            "revision": "719aaef8225945c0d80b277de6c79aa42ab053d5",
+        },
+        type="AudioClassification",
+        category="a2t",
+        eval_splits=["train", "test"],
```
Comment (isaac-chung) [inline mteb/tasks/Audio/AudioClassification/eng/VoxPopuliAccentID.py]:
```suggestion
        eval_splits=["test"],
```

File: mteb/tasks/Audio/AudioClassification/eng/VoxPopuliAccentID.py [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+    )
+
+    audio_column_name: str = "audio"
+    label_column_name: str = "accent"
+    samples_per_label: int = 50
+    is_cross_validation: bool = False
+
+    def dataset_transform(self):
+        # Split test into train (80%) and new test (20%)
+        import random
```
Comment (isaac-chung) [inline mteb/tasks/Audio/AudioClassification/eng/VoxPopuliAccentID.py]: Any reason to do this extra split, when the test split is readily available?

File: mteb/tasks/Audio/AudioClassification/eng/VoxPopuliGenderID.py [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+                "train": 7600,
+                "validation": 1750,
+                "test": 1840,
+            },
+        },
+    )
+
+    audio_column_name: str = "audio"
+    label_column_name: str = "gender"
+    samples_per_label: int = 100
```
Comment (isaac-chung) [inline mteb/tasks/Audio/AudioClassification/eng/VoxPopuliGenderID.py]: Can we set samples_per_label at a lower value, like 16?

File: mteb/tasks/Audio/AudioClassification/eng/VoxPopuliLanguageID.py [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+            "n_samples": {
+                "train": 6200,  # ~80% of test examples
+                "test": 1600,  # ~20% of test examples
+            },
+        },
+    )
+
+    audio_column_name: str = "audio"
+    label_column_name: str = "language"
+    samples_per_label: int = 50  # For balanced training
```
Comment (isaac-chung) [inline mteb/tasks/Audio/AudioClassification/eng/VoxPopuliLanguageID.py]: Can we set samples_per_label at a lower value, like 16?

**Review by isaac-chung — COMMENTED** [review]
Got a few questions below.

**Comment by AdnanElAssadi** [comment]
**Model: "microsoft/wavlm-base"**

```python
{
  "dataset_revision": "719aaef8225945c0d80b277de6c79aa42ab053d5",
  "task_name": "VoxPopuliAccentClustering",
  "mteb_version": "1.34.7",
  "scores": {
    "test": [
      {
        "v_measure": 0.075173,
        "nmi": 0.075173,
        "ari": 0.01977,
        "cluster_accuracy": 0.166567,
        "main_score": 0.166567,
        "hf_subset": "default",
        "languages": [
          "eng-Latn"
        ]
      }
    ]
  },
  "evaluation_time": 176.5011546611786,
  "kg_co2_emissions": null
}
```

**Model: "MIT/ast-finetuned-audioset-10-10-0.4593"**

```python

{
  "dataset_revision": "719aaef8225945c0d80b277de6c79aa42ab053d5",
  "task_name": "VoxPopuliAccentID",
  "mteb_version": "1.34.7",
  "scores": {
    "train": [
      {
        "accuracy": 0.22319,
        "f1": 0.212795,
        "f1_weighted": 0.229425,
        "scores_per_experiment": [
          {
            "accuracy": 0.229851,
            "f1": 0.214561,
            "f1_weighted": 0.235229
          },
          {
            "accuracy": 0.239403,
            "f1": 0.229795,
            "f1_weighted": 0.246091
          },
          {
            "accuracy": 0.205373,
            "f1": 0.198601,
            "f1_weighted": 0.214797
          },
          {
            "accuracy": 0.223284,
            "f1": 0.216803,
            "f1_weighted": 0.224414
          },
          {
            "accuracy": 0.218041,
            "f1": 0.204215,
            "f1_weighted": 0.226593
          }
        ],
        "main_score": 0.22319,
        "hf_subset": "default",
        "languages": [
          "eng-Latn"
        ]
      }
    ]
  },
  "evaluation_time": 896.4413390159607,
  "kg_co2_emissions": null
}
```

**Comment by AdnanElAssadi** [comment]
@isaac-chung Regarding your comments about using the test set rather than the training set, you're right for clustering. For classification, however, cross validation only works with a training set being specified as the eval split. I did a workaround above by copying the test set to training to do what you mentioned and technically evaluate on test.

I managed to get results for the clustering and classification accent subsets of VoxPopuli. I am not sure if others will have to be downsampled, but they currently are taking quite a bit of time.

**Comment by isaac-chung** [comment]
Thanks. In that case, let's try the following:

1. Turn cross validation off and use only the test splits
2. Downsample the test split to 2048 using `self.stratified_subsampling` (from AbsTask) if it exceeds 2048 rows
3. Use samples_per_label to 8 instead of 30 or 50

**Comment by AdnanElAssadi** [comment]
> Thanks. In that case, let's try the following:
> 
> 1. Turn cross validation off and use only the test splits
> 2. Downsample the test split to 2048 using `self.stratified_subsampling` (from AbsTask) if it exceeds 2048 rows
> 3. Use samples_per_label to 8 instead of 30 or 50

1. What about the training split? From my understanding, turning cross validation on means we should have both train and test splits for each.
2. You mean as a preprocessing step? Does that subsample the dataset and push to HF? I usually do it externally.
3. I think the main issue is downloading the huge dataset initially and not the actual evaluation of 30 vs 8 after download, but not sure. Anyway though, we can try with 8.

**Comment by isaac-chung** [comment]
1. Specifying train in eval_splits means to eval on the train set, and we don't want that. In this AbsTask, the train set is already being read.
2. It does not. Often in MMTEB we simply use it as a preprocessing step. @KennethEnevoldsen what's your take on uploading a subsampled set and using that instead of the origin dataset? I see it's done for birdclef at the moment.
3. Got it. And my bad - the default is 16

**Comment by KennethEnevoldsen** [comment]
2) I would def. upload a default sized dataset that it not super large.

Recently had a talk with an embedding company and it seems like a big reason for them avoiding a benchmark is compute cost - if it is too big they simply don't want to touch it as they don't want to spend the time getting to know the benchmark to then figure out a good way to downsample. Default should be as small as possible but as large as required for a good estimate

**Comment by KennethEnevoldsen** [comment]
@AdnanElAssadi56 any status on this dataset?

**Comment by AdnanElAssadi** [comment]
> @AdnanElAssadi56 any status on this dataset?

Yes, I’ve tried using truncation with the models, which helped solve the issue of extremely long audio samples bottlenecking execution, similar to what we encountered in BirdClef. However, in this case of Voxpopuli, the main bottleneck is actually the dataset download itself, so truncation didn’t address the core issue.

I’ll try downsampling the dataset again, but the main challenge I’ve faced with downsampling it previously is that it typically requires downloading the entire dataset to disk before preprocessing, which is quite resource-intensive.

**Comment by KennethEnevoldsen** [comment]
A solution might be to stream it in?

**Comment by AdnanElAssadi** [comment]
> A solution might be to stream it in?

I'll try. I thought it doesn't preserve audio format, but I'll check.

**Comment by AdnanElAssadi** [comment]
**Model: MIT/ast-finetuned-audioset-10-10-0.4593**


```python

{
  "dataset_revision": "70031eb5affcb0805e448fdf0b2dbbfc05f0aa8f",
  "task_name": "VoxPopuliGenderID",
  "mteb_version": "1.34.7",
  "scores": {
    "train": [
      {
        "accuracy": 0.868,
        "f1": 0.864686,
        "f1_weighted": 0.868421,
        "ap": 0.81358,
        "ap_weighted": 0.81358,
        "scores_per_experiment": [
          {
            "accuracy": 0.9,
            "f1": 0.897373,
            "f1_weighted": 0.900985,
            "ap": 0.800057,
            "ap_weighted": 0.800057
          },
          {
            "accuracy": 0.88,
            "f1": 0.875,
            "f1_weighted": 0.8805,
            "ap": 0.880225,
            "ap_weighted": 0.880225
          },
          {
            "accuracy": 0.76,
            "f1": 0.755202,
            "f1_weighted": 0.760685,
            "ap": 0.630022,
            "ap_weighted": 0.630022
          },
          {
            "accuracy": 0.88,
            "f1": 0.875982,
            "f1_weighted": 0.88,
            "ap": 0.866952,
            "ap_weighted": 0.866952
          },
          {
            "accuracy": 0.92,
            "f1": 0.919872,
            "f1_weighted": 0.919936,
            "ap": 0.890643,
            "ap_weighted": 0.890643
          }
        ],
        "main_score": 0.868,
        "hf_subset": "default",
        "languages": [
          "eng-Latn",
          "fra-Latn",
          "spa-Latn",
          "pol-Latn",
          "deu-Latn"
        ]
      }
    ]
  },
  "evaluation_time": 84.80226755142212,
  "kg_co2_emissions": null
}

```


```python

{
  "dataset_revision": "70031eb5affcb0805e448fdf0b2dbbfc05f0aa8f",
  "task_name": "VoxPopuliLanguageID",
  "mteb_version": "1.34.7",
  "scores": {
    "train": [
      {
        "accuracy": 0.378,
        "f1": 0.370261,
        "f1_weighted": 0.384193,
        "scores_per_experiment": [
          {
            "accuracy": 0.44,
            "f1": 0.42462,
            "f1_weighted": 0.464762
          },
          {
            "accuracy": 0.33,
            "f1": 0.326178,
            "f1_weighted": 0.328191
          },
          {
            "accuracy": 0.4,
            "f1": 0.381915,
            "f1_weighted": 0.395243
          },
          {
            "accuracy": 0.34,
            "f1": 0.335237,
            "f1_weighted": 0.3442
          },
          {
            "accuracy": 0.38,
            "f1": 0.383352,
            "f1_weighted": 0.388567
          }
        ],
        "main_score": 0.378,
        "hf_subset": "default",
        "languages": [
          "eng-Latn",
          "fra-Latn",
          "spa-Latn",
          "pol-Latn",
          "deu-Latn"
        ]
      }
    ]
  },
  "evaluation_time": 82.0283854007721,
  "kg_co2_emissions": null
}

```


```python
{
  "dataset_revision": "70031eb5affcb0805e448fdf0b2dbbfc05f0aa8f",
  "task_name": "VoxPopuliGenderClustering",
  "mteb_version": "1.34.7",
  "scores": {
    "train": [
      {
        "v_measure": 0.000298,
        "nmi": 0.000298,
        "ari": -0.001558,
        "cluster_accuracy": 0.51,
        "main_score": 0.51,
        "hf_subset": "default",
        "languages": [
          "eng-Latn",
          "fra-Latn",
          "spa-Latn",
          "pol-Latn",
          "deu-Latn"
        ]
      }
    ]
  },
  "evaluation_time": 44.37592005729675,
  "kg_co2_emissions": null
}
```

**Comment by AdnanElAssadi** [comment]
@KennethEnevoldsen @isaac-chung 

I’ve subsampled the dataset using a stratified approach based on both language and gender.  You can check the subsampled dataset on my HF. I've also included the 5 multilingual languages as @isaac-chung previously suggested.

I got the above results on the AST Model.

**Comment by isaac-chung** [comment]
Thanks again, @AdnanElAssadi56 !
